magenta-lib/src/main/scala/magenta/graph/Graph.scala

```diff
@@ -74,29 +80,49 @@ case class Graph[T](edges: Set[Edge[T]]) {
   }
 
   def joinSeries(other: Graph[T]): Graph[T] = {
```

Comment:
This was pretty broken in the way it modified (or didn't) priorities when gluing graphs together end to end. It feels like there should be a simpler way to express this, but it is at least passing a suite of tests.